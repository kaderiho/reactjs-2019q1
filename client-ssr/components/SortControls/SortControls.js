"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Button = _interopRequireDefault(require("../Button/Button"));

var _sort = require("../../store/actions/sort");

var _reactRedux = require("react-redux");

var SortControls = function SortControls(_ref) {
  var sortBy = _ref.sortBy,
      onSortByHandler = _ref.onSortByHandler;
  return _react.default.createElement("div", {
    className: "sortControls"
  }, _react.default.createElement("span", {
    className: "sortControls-label"
  }, "Sort by"), _react.default.createElement(_Button.default, {
    className: (0, _classnames.default)("sortControls-control", {
      'sortControls-control--active': sortBy === 'release_date'
    }),
    onClick: function onClick() {
      onSortByHandler('release_date');
    },
    color: "transparent",
    size: "small"
  }, "Release date"), _react.default.createElement(_Button.default, {
    className: (0, _classnames.default)("sortControls-control", {
      'sortControls-control--active': sortBy === 'rating'
    }),
    onClick: function onClick() {
      onSortByHandler('rating');
    },
    color: "transparent",
    size: "small"
  }, "Rating"));
};

var mapStateToProps = function mapStateToProps(_ref2) {
  var sortBy = _ref2.sortBy;
  return {
    sortBy: sortBy
  };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
  return {
    onSortByHandler: function onSortByHandler(sortBy) {
      dispatch((0, _sort.SET_SORT_BY)(sortBy));
    }
  };
};

var _default = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(SortControls);

exports.default = _default;