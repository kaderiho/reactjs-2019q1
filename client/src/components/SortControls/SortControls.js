import React from 'react';
import classnames from 'classnames';
import Button from 'components/Button/Button';
import './SortControls.scss';

import { SET_SORT_BY } from 'store/actions/sort';
import { connect } from 'react-redux';

let SortControls = ({ sortBy, onSortByHandler }) => (
    <div className="sortControls">
        <span className="sortControls-label">Sort by</span>

        <Button
            className={classnames("sortControls-control", {
                'sortControls-control--active' : sortBy === 'release_date'
            })}
            onClick={() => { onSortByHandler('release_date') }}
            color="transparent"
            size="small">
            Release date
        </Button>

        <Button
            className={classnames("sortControls-control", {
                'sortControls-control--active' : sortBy === 'rating'
            })}
            onClick={() => { onSortByHandler('rating') }}
            color="transparent"
            size="small">
            Rating
        </Button>
    </div>
);

const mapStateToProps = ({ sortBy }) => ({
    sortBy
});

const mapDispatchToProps = dispatch => ({
    onSortByHandler(sortBy) {
        dispatch(SET_SORT_BY(sortBy));
    }
});

SortControls = connect(mapStateToProps, mapDispatchToProps)(SortControls);

export default SortControls;