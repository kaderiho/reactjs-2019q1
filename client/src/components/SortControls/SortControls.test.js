import React from 'react';
import { shallow } from 'enzyme';

import SortControls from './SortControls';

/**
 * Tests
 * 
 * - To match snapshot
 * - TODO: check 'sortControls-control--active' when filter button is clicked
 */
describe('<SortControls /> component', () => {
    it('Should render correctly', () => {
        const wrapper = shallow(<SortControls />);

        expect(wrapper).toMatchSnapshot();
    });
});