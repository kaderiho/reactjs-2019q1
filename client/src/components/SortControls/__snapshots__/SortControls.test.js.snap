// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortControls /> component Should render correctly 1`] = `
<div
  className="sortControls"
>
  <span
    className="sortControls-label"
  >
    Sort by
  </span>
  <Button
    className="sortControls-control sortControls-control--active"
    color="transparent"
    size="small"
  >
    Release data
  </Button>
  <Button
    className="sortControls-control"
    color="transparent"
    size="small"
  >
    Rating
  </Button>
</div>
`;
